agenta-web/src/components/EvaluationTable/SimilarityMatchEvaluationTable.tsx

```diff
@@ -0,0 +1,388 @@
+        input_value: string
+    }[]
+    outputs: {
+        variant_name: string
+        variant_output: string
+    }[]
+    columnData0: string
+    correctAnswer: string
+    score: string
+    evaluationFlow: EvaluationFlow
+}
+/**
+ *
+ * @param appEvaluation - Evaluation object
+ * @param evaluationRows - Evaluation rows
+ * @param columnsCount - Number of variants to compare face to face (per default 2)
+ * @returns
+ */
+
+const SimilarityMatchEvaluationTable: React.FC<SimilarityMatchEvaluationTableProps> = ({
+    appEvaluation,
+    evaluationRows,
+    columnsCount,
+}) => {
+    const router = useRouter()
+    let app_name = ""
+    if (Array.isArray(router.query.app_name)) {
+        app_name = router.query.app_name[0]
+    } else if (typeof router.query.app_name === "string") {
+        app_name = router.query.app_name
+    }
+    const variants = appEvaluation.variants
+
+    const variantData = variants.map((variant: Variant) => {
+        const {optParams, URIPath, isLoading, isError, error} = useVariant(app_name, variant)
+
+        return {
+            optParams,
+            URIPath,
+            isLoading,
+            isError,
+            error,
+        }
+    })
+
+    const [rows, setRows] = useState<SimilarityMatchEvaluationTableRow[]>([])
+    const [dissimilarAnswers, setDissimilarAnswers] = useState<number>(0)
+    const [similarAnswers, setSimilarAnswers] = useState<number>(0)
+    const [accuracy, setAccuracy] = useState<number>(0)
+
+    useEffect(() => {
+        if (evaluationRows) {
+            setRows(evaluationRows)
+        }
+    }, [evaluationRows])
+
+    useEffect(() => {
+        if (similarAnswers + dissimilarAnswers > 0) {
+            setAccuracy((similarAnswers / (similarAnswers + dissimilarAnswers)) * 100)
+        } else {
+            setAccuracy(0)
+        }
+    }, [similarAnswers, dissimilarAnswers])
+
+    useEffect(() => {
+        const similar = rows.filter((row) => row.score === "true").length
+        const dissimilar = rows.filter((row) => row.score === "false").length
+        const accuracy = similar + dissimilar > 0 ? (similar / (similar + dissimilar)) * 100 : 0
+
+        setSimilarAnswers(similar)
+        setDissimilarAnswers(dissimilar)
+        setAccuracy(accuracy)
+    }, [rows])
+
+    const handleInputChange = (
+        e: React.ChangeEvent<HTMLInputElement>,
+        rowIndex: number,
+        inputFieldKey: number,
+    ) => {
+        const newRows = [...rows]
+        newRows[rowIndex].inputs[inputFieldKey].input_value = e.target.value
+        setRows(newRows)
+    }
+
+    const runAllEvaluations = async () => {
+        const promises: Promise<void>[] = []
+
+        for (let i = 0; i < rows.length; i++) {
+            promises.push(runEvaluation(i))
+        }
+
+        Promise.all(promises)
+            .then(() => {
+                console.log("All functions finished.")
+            })
+            .catch((err) => console.error("An error occurred:", err))
```

Comment:
**Suggestion:** Avoid using `console.error` for error handling. Consider using a proper error handling mechanism or logging library to handle and log errors.
```suggestion
            catch((err) => {
                // Implement proper error handling here
                // console.error("An error occurred:", err)
            })
```